src/components/LinkToDepositPage.tsx

```diff
@@ -0,0 +1,15 @@
+import React from 'react'
+import { useLocation } from 'react-router'
+import { Link } from 'react-router-dom'
+
+const LinkToDepositPage: React.FC = ({ children }) => {
+  const location = useLocation()
+  const to = {
```

Comment:
I'm not sure it's a good idea to create a whole component just for `ToDepositPage` use case.
Too specific, and we use it only a couple times.
If we really need a `LinkWithPastLocation` or something, that is another matter.
Also should pass [all the possible props along](https://reacttraining.com/react-router/web/api/Link/others)

```js
const LinkWithPastLoc = (props) => {
  const location = useLocation()
  const to = {...props.to, state : {...props.to?.state, from: location}}
  return <Link {...props} to={to}/>
}
```

Reply:
Can then use as:
```xml
<LinkWithPastLoc to="/deposit">Exchange wallet:</Link>
```

Reply:
I wouldn't shorten the words:  `LinkWithPastLocation` or even `LinkWithFromLocation`

Reply:
I can't make TS happy with your proposal @Velenir.
I pushed some changes that looks kinda like that, but not quite.
It works, but doesn't look nice. Help?

Reply:
Have a look at #236

Reply:
Thank you! Merged changes into this PR